src/main/scala/io/iohk/ethereum/jsonrpc/JsonRpcController.scala

```diff
@@ -0,0 +1,52 @@
+package io.iohk.ethereum.jsonrpc
+
+import io.iohk.ethereum.jsonrpc.JsonRpcController.{JsonDecoder, JsonEncoder}
+import io.iohk.ethereum.jsonrpc.Web3Service.{ClientVersionRequest, ClientVersionResponse, Sha3Request, Sha3Response}
+import org.json4s.JsonAST.{JArray, JValue}
+import org.json4s.JsonDSL._
+
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext.Implicits.global
+
+object JsonRpcController {
+  trait JsonDecoder[T] {
+    def decodeJson(params: Option[JArray]): Either[JsonRpcError, T]
+  }
+
+  trait JsonEncoder[T] {
+    def encodeJson(t: T): JValue
+  }
+}
+
+class JsonRpcController(web3Service: Web3Service) {
+
+  import JsonMethodsImplicits._
+  import JsonRpcErrors._
+
+  def handleRequest(request: JsonRpcRequest): Future[JsonRpcResponse] = {
+    request.method match {
```

Comment:
I think this should be mixed from different handlers. I'm thinking about this in order to disable different apis. This is not a blocker tho, it can be refactored later.